web-ng/src/app/services/layer/layer.service.ts

```diff
@@ -76,10 +81,10 @@ export class LayerService {
   }
 
   /**
-   * Updates layer of a project with a given layer value.
+   * Adds/Updates the layer of a project with a given layer value.
    */
-  updateLayer(projectId: string, layer: Layer): Promise<void> {
-    return this.dataStoreService.updateLayer(projectId, layer);
+  addOrUpdateLayer(projectId: string, layer: Layer): Promise<void> {
+    return this.dataStoreService.addOrUpdateLayer(projectId, layer);
```

Comment:
I would suggest making `getLayerCount()` private and updating the index here if the layer is a new one. We can tell if the layer is new by using a special value like '-1' until this method is called. This will abstract the details of how the layer is added and ensure the index is set right before saving.

Reply:
Thanks , I really like this point 👍